### PR TITLE
Remove old flutter messaging API

### PR DESCRIPTION
**Pull request by mravn-google**
Breaking change: removed facilities for JSON and string messaging from FlutterView/FlutterViewController, leaving only binary messaging there. All other use of flutter communication now goes through FlutterMessageChannel and FlutterMethodChannels. Retained use of String and JSON codecs for now.

Companion flutter PR: https://github.com/flutter/flutter/pull/8837

### CODE REVIEW

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 1]
```diff
@@ -189,77 +209,49 @@ public void addActivityLifecycleListener(ActivityLifecycleListener listener) {
...
-        }
+        mFlutterLocalizationChannel.invokeMethod("setLocale",
+            Arrays.asList(locale.getLanguage(), locale.getCountry()));
     }
 
     @Override
     protected void onConfigurationChanged(Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
         setLocale(newConfig.locale);
+        Log.e(TAG, "config changed");
```
Comment (abarth) [inline shell/platform/android/io/flutter/view/FlutterView.java]: ?

**Comment by abarth** [comment]
LGTM

**Comment by eseidelGoogle** [comment]
FYI this will sign you up for a bit of work when we roll into Google3 next time. :D  FYI @tvolkert

**Comment by tvolkert** [comment]
@goderbauer already has a CL in progress for when we update to SDK 1.23 internally

**Pull request merged**